admin_manual/installation/nginx.rst

```diff
@@ -282,75 +262,89 @@ In this example the webroot is located at
           gzip_proxied expired no-cache no-store private no_last_modified no_etag auth;
           gzip_types application/atom+xml application/javascript application/json application/ld+json application/manifest+json application/rss+xml application/vnd.geo+json application/vnd.ms-fontobject application/x-font-ttf application/x-web-app-manifest+json application/xhtml+xml application/xml font/opentype image/bmp image/svg+xml image/x-icon text/cache-manifest text/css text/plain text/vcard text/vnd.rim.location.xloc text/vtt text/x-component text/x-cross-domain-policy;
 
-          # Uncomment if your server is build with the ngx_pagespeed module
-          # This module is currently not supported.
+          # Pagespeed is not supported by Nextcloud, so if your server is built
+          # with the `ngx_pagespeed` module, uncomment this line to disable it.
           #pagespeed off;
-
-          location /nextcloud {
-              rewrite ^ /nextcloud/index.php;
-          }
-
-          location ~ ^\/nextcloud\/(?:build|tests|config|lib|3rdparty|templates|data)\/ {
-              deny all;
+          
+          # HTTP response headers borrowed from Nextcloud `.htaccess`
+          add_header Referrer-Policy                      "no-referrer"   always;
+          add_header X-Content-Type-Options               "nosniff"       always;
+          add_header X-Download-Options                   "noopen"        always;
+          add_header X-Frame-Options                      "SAMEORIGIN"    always;
+          add_header X-Permitted-Cross-Domain-Policies    "none"          always;
+          add_header X-Robots-Tag                         "none"          always;
+          add_header X-XSS-Protection                     "1; mode=block" always;
+          
+          # Remove X-Powered-By, which is an information leak
+          fastcgi_hide_header X-Powered-By;
+          
+          # Path to the root of your installation
+          # Notice that this is an `alias` directive, not a `root` directive
+          alias /var/www/nextcloud;
```

Comment:
I've tested the subdir setup, and it turns out that using `alias` is problematic. We can either:

1. Switch back to using `root`, e.g. `root /var/www` like in the currently published config, but this means the name of the Nextcloud root directory must match that of the URI, e.g. `nextcloud`, which may be undesireable; or

2. Keep `alias`, and instead:

  a. change `fastcgi_param SCRIPT_FILENAME` from `$document_root$fastcgi_script_name` to `$request_filename` in the `~\.php$` block; and

  b. change all instances of `/nextcloud/index.php$request_uri` to just `/nextcloud/index.php, and add `fastcgi_param REQUEST_URI $request_uri` instead. Strangely, I faced issues with Nextcloud not handling requests properly when using the former method, but the latter works fine.

I think we should go with (1) for simplicity, but the behaviour I've seen regarding `$request_uri` suggests that perhaps we should be using `fastcgi_param REQUEST_URI $request_uri` in both configs, rather than appending `$request_uri` to `/index.php`.

Thoughts?

Reply:
I think ideally we should go to a model that uses `alias`, since some users will want to host their application using a different webdir (like **nc** or **mycloud**). But for now, if you just want to switch back to `root` that should be fine since that is how it was. That will allow us to approve these change and not hold it up for something somewhat unrelated.

Then if you or I or someone else wants to tackle using `alias` we can just create a new PR.

Reply:
Changes made, feel free to review now

Reply:
Everything looks fine to me, my production system is basically using this (subdir using `root`).
Since you started these 2 comments I'll let you resolve them, https://github.com/nextcloud/documentation/pull/2197#pullrequestreview-462693331 https://github.com/nextcloud/documentation/pull/2197#pullrequestreview-463714625.

But otherwise I think these changed are all good.